pdns/syncres.cc

```diff
@@ -189,6 +337,10 @@ int SyncRes::beginResolve(const DNSName &qname, const QType qtype, QClass qclass
     }
   }
 
+  auto luaLocal = g_luaconfs.getLocal();
+  if (res == 0 && qclass == QClass::IN && luaLocal->allowAdditionalQTypes.find(qtype) != luaLocal->allowAdditionalQTypes.end()) {
```

Comment:
It looks like we are going to do the `luaLocal->allowAdditionalQTypes.find(qtype)` in the function anyway, so we might want to skip it here?

Reply:
It is an optimization to not setup all the sets etc if we know beforehand there will be no results.